### ISSUE TITLE
[Bug Report] ttnn.embedding crashes if accidentally pass TILED tensor as input

### ISSUE THREAD

**Issue by dmakoviichuk-tt** [tenstorrent/tt-metal#13657]
**Describe the bug**

**To Reproduce**
call ttnn.embedding with wrong tensor layout.

**Expected behavior**
In perfect world it would be nice to have support for TILED_LAYOUT.
embedding_bw returns TILED gradients. Right now to deal with we store our weights as TILED but untilize them each call:
https://github.com/tenstorrent/TT-Tron/blob/main/sources/ttml/ops/embedding_op.cpp

**Additional context**
Feel free to ask any questions about our implementations. Or on how to build our code.

**Comment by dmakoviichuk-tt** [comment]
@davorchap  it is must have for the BPE Tokenizers.

**Comment by ntarafdar** [comment]
Hey @yugi957 was this case missed with the embedding pr?

**Comment by yugi957** [comment]
Not 100% sure I understand, in this bug the tensor accuracy is correct but the output layout is incorrect?

**Comment by dmakoviichuk-tt** [comment]
@yugi957  I might missed that it was already fixed but embedding requires anly row_major as input and we need to have tiled weights. But embedding matrices could be really huge and we cannot afford having tiled -> row_major conversion for each training step. And we cannot always have this weights for a different resons (backward version returns only tiled gradients and we need to use the same embedding matrix as the last neural network layer.).
If you can update embedding op kernel so it can take tiled weights as input it would reduce memory consumption and improve our performance.

**Comment by davorchap** [comment]
> [@yugi957](https://github.com/yugi957) I might missed that it was already fixed but embedding requires anly row_major as input and we need to have tiled weights. But embedding matrices could be really huge and we cannot afford having tiled -> row_major conversion for each training step. And we cannot always have this weights for a different resons (backward version returns only tiled gradients and we need to use the same embedding matrix as the last neural network layer.). If you can update embedding op kernel so it can take tiled weights as input it would reduce memory consumption and improve our performance.

@yugi957 , the embedding OP can take tilied inputs and untilize on the fly, many OPs do that.

**Comment by ntarafdar** [comment]
@dmakoviichuk-tt , @yugi957 is going to try this. He has unit tests that has tiled weights but not tiled indices at the moment.
This should work though.

So that we can be consistent with what he has in his unit tests could you give him the shape of your inputs for him to test.

**Comment by ntarafdar** [comment]
What we found out about this is that there is a bug in untilize with uint32 inputs still :( .
@yugi957 can you comment if uint16 indices worked.

We will track this with a fix in untilize

**Comment by ntarafdar** [comment]
@dmakoviichuk-tt can you comment on data type and shapes that you have for embedding. It looks like an untilize bug outside the scope of embedding.

**Comment by prajaramanTT** [comment]
@dmakoviichuk-tt Can you add priority to this ticket ?

**Comment by prajaramanTT** [comment]
@dmakoviichuk-tt @ntarafdar Is this still an open issue ? If not, can you please close this ticket ? Thanks.

**Comment by yugi957** [comment]
For specifically tiled inputs, this issue should be closeable, but I'll wait for the issuer to make sure
